distributed/client.py

```diff
@@ -1053,6 +1053,14 @@ def _update_scheduler_info(self):
         except EnvironmentError:
             logger.debug("Not able to query scheduler for identity")
 
+    @gen.coroutine
+    def _wait_until_n_workers(self, n):
+        while n and len(self.cluster.scheduler.workers) < n:
+            yield gen.sleep(0.01)
```

Comment:
I agree with @guillaumeeb 's point that we shouldn't assume the existence of either the scheduler or even the cluster attribute.  The generic way to do this is to poll the scheduler's identity function.

```python
while True:
    info = yield self.scheduler.identity()
    if len(info['workers']) >= n:
        break
    else:
        yield gen.sleep(0.200)
```

The `self.scheduler` attribute used above is a connection to the scheduler that will always be there, rather than an explicit reference to the scheduler itself.